src/main/java/com/northconcepts/datapipeline/foundations/examples/pipeline/StoreDatasetInMemory.java

```diff
@@ -0,0 +1,24 @@
+package com.northconcepts.datapipeline.foundations.examples.pipeline;
+
+import com.northconcepts.datapipeline.csv.CSVReader;
+import com.northconcepts.datapipeline.foundations.pipeline.Pipeline;
+import com.northconcepts.datapipeline.foundations.pipeline.dataset.Dataset;
+import com.northconcepts.datapipeline.foundations.pipeline.dataset.MemoryDataset;
+
+import java.io.File;
+
+public class StoreDatasetInMemory {
+
+    public static void main(String[] args) {
+        Pipeline pipeline = new Pipeline()
+                .setInputAsDataReaderFactory(() -> new CSVReader(new File("example/data/input/credit-balance-01.csv"))
+                        .setFieldNamesInFirstRow(true));
+        
+        Dataset dataset = new MemoryDataset(pipeline);
+        
+        dataset.load().waitForRecordsToLoad();
```

Comment:
Can we show some data from the dataset after we wait?

The `Dataset` is `Iterable` so just use it in an enhanced for loop.

Reply:
Done